### PR TITLE
docs: update documentation for Talos 1.4

### PR DESCRIPTION
**Pull request by smira**
Updated documentation, what's new, etc.

Also fix some minor UI issues in the dashboard.

### CODE REVIEW

File: website/content/v1.4/introduction/what-is-new/index.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+Talos now supports loading network configuration on VMWare platform from the `metadata` key.
+See [CAPV IPAM Support](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/proposal/20220929-ipam-support.md) and
+[Talos issue 6708](https://github.com/siderolabs/talos/issues/6708) for details.
+
+## Component Updates
+
+* Linux: 6.1.23
+* containerd: v1.6.20
+* runc: v1.1.5
+* Kubernetes: v1.27.0-rc.1
```
Comment (frezbo) [inline website/content/v1.4/introduction/what-is-new/index.md:165]: didn't we update to 1.27?

File: website/content/v1.4/introduction/what-is-new/index.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+* etcd: v3.5.7
```
Comment (frezbo) [inline website/content/v1.4/introduction/what-is-new/index.md:166]: 3.5.8

**Review by frezbo — APPROVED** [review]

File: website/content/v1.4/introduction/what-is-new/index.md (hunk 2 above)
Comment (smira) [reply website/content/v1.4/introduction/what-is-new/index.md:166]: thanks

File: website/content/v1.4/introduction/support-matrix.md [hunk 3]
```diff
@@ -6,8 +6,8 @@ description: "Table of supported Talos Linux versions and respective platforms."
...
-| End of Community Support                                                                                       | 1.5.0 release (2023-06-15, TBD)    | 1.3.0 release (2022-12-15)         |
+| Release Date                                                                                                   | 2023-04-18                         | 2022-12-15 (1.3.0)                 |
+| End of Community Support                                                                                       | 1.5.0 release (2023-08-15)         | 1.4.0 release (2023-04-18)         |
 | Enterprise Support                                                                                             | [offered by Sidero Labs Inc.](https://www.siderolabs.com/support/) | [offered by Sidero Labs Inc.](https://www.siderolabs.com/support/) |
 | Kubernetes                                                                                                     | 1.27, 1.26, 1.22                   | 1.26, 1.25, 1.24                   |
```
Comment (frezbo) [inline website/content/v1.4/introduction/support-matrix.md:12]: the k8s version here should be `1.27, 1.26, 1.25`

Comment (smira) [reply website/content/v1.4/introduction/support-matrix.md:12]: yes, thanks for noticing that!

**Pull request merged**